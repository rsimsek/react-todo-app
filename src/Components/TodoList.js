import React from "react";
import Todo from "./Todo";

const TodoList = ({ todos, toggleTodo }) => {
  const todoNode = todos.map(todo => {
    return <Todo todo={todo} key={todo.id} toggleTodo={toggleTodo} />;
  });

  return (
    <ul>
      {todoNode}
    </ul>
  );
};

export default TodoList;
