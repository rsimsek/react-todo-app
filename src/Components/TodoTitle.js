import React from "react";

const TodoTitle = (props) => (
      <h2>{props.title}</h2>
);

export default TodoTitle;
