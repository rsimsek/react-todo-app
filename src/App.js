import React, { Component } from "react";
import TodoTitle from "./Components/TodoTitle";
import TodoForm from "./Components/TodoForm";
import TodoList from "./Components/TodoList";
import "./App.css";

//const todos = [{id : 1, value: 'Learn React', completed: true}, {id : 2, value: 'Learn Redux', completed: false}];

window.id = 1;
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      todos: []
    };
  }

  addTodo = val => {
    // Assemble data
    const todo = { text: val, id: window.id++, isCompleted: false };
    if (!todo.text) console.log("shit");
    // Update data
    this.state.todos.push(todo);
    // Update state
    this.setState({ todos: this.state.todos });
    //console.log(this.state.todos);
  };

  toggleTodo = id => {
    this.state.todos.map(todo => {
      if (todo.id === id) return (todo.isCompleted = !todo.isCompleted);
      else return null;
    });
    this.setState({ todos: this.state.todos });
  };

  render() {
    return (
      <div style={{ App }}>
        <TodoTitle />
        <TodoForm addTodo={this.addTodo} />
        <TodoList todos={this.state.todos} toggleTodo={this.toggleTodo} />
      </div>
    );
  }
}

export default App;
