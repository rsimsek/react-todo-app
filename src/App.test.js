import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import TodoTitle from './Components/TodoTitle';
import TodoList from './Components/TodoList';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
});

test('renders todo title', () => {
  const Title = 'Test Title';
  const component = renderer.create(
    <TodoTitle title={Title}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('renders list of todos', () => {
  const todos = [{id : 1, value: 'Learn React', completed: true}, {id : 2, value: 'Learn Redux', completed: false}];
  const component = renderer.create(
    <TodoList todos={todos} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})