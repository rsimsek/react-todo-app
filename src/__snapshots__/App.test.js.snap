// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders list of todos 1`] = `
<ul>
  <li
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  />
  <li
    onClick={[Function]}
    style={
      Object {
        "textDecoration": "none",
      }
    }
  />
</ul>
`;

exports[`renders todo title 1`] = `
<h2>
  Test Title
</h2>
`;
